core/vm/contracts.go

```diff
@@ -34,12 +34,16 @@ import (
 	"golang.org/x/crypto/ripemd160"
 )
 
+type StateReader interface {
+	GetState(common.Address, common.Hash) common.Hash
+}
+
 // PrecompiledContract is the basic interface for native Go contracts. The implementation
 // requires a deterministic gas count based on the input size of the Run method of the
 // contract.
 type PrecompiledContract interface {
-	RequiredGas(input []byte) uint64  // RequiredPrice calculates the contract gas use
-	Run(input []byte) ([]byte, error) // Run runs the precompiled contract
+	RequiredGas(input []byte) uint64                       // RequiredPrice calculates the contract gas use
+	Run(statedb StateReader, input []byte) ([]byte, error) // Run runs the precompiled contract
```

Comment:
Nitpick, we have `statedb` and `stateDB` naming in this single PR. Can we just have `state` instead?